src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -689,49 +863,202 @@ private static IEnumerable<double> LazySequence(double records, double start, do
             }
         }
 
-        public static FormulaValue Abs(IRContext irContext, NumberValue[] args)
+        private static IEnumerable<decimal> LazySequenceDecimal(int records, decimal start, decimal step)
+        {
+            var x = start;
+            for (var i = 1; i <= records; i++)
+            {
+                yield return x;
+                x += step;

```

Comment:
>x += step [](http://example.com/codeflow?start=16&length=9)

could this overflow?

Reply:
Great call!  Yes, this could overflow, and I think the right thing to do would be to include an ErrorValue in this situation.  It's easier to hit with Decimal but has the same problem in Float.  I'm going to punt this for the moment and look for the best way to address both.  https://github.com/microsoft/Power-Fx/issues/1368